src/core/proj/qgscoordinatetransform_p.cpp

```diff
@@ -158,6 +159,10 @@ bool QgsCoordinateTransformPrivate::initialize()
     return true;
   }
 
+  mGeographicToWebMercator =
+    mSourceCRS.authid() == QLatin1String( "EPSG:4326" ) &&
```

Comment:
I'm guessing this fix also applies to non-wgs84 geographic source CRS?

```suggestion
    mSourceCRS.isGeographic() &&
```

Reply:
> I'm guessing this fix also applies to non-wgs84 geographic source CRS?

yes for what we do here with this flag, that should be OK